app/models/lookup_key.rb

```diff
@@ -214,7 +207,7 @@ def array2path(array)
   def validate_and_cast_default_value
     return true if default_value.nil? || contains_erb?(default_value)
     begin
-      self.default_value = cast_validate_value self.default_value
+      Foreman::Parameters::Caster.new(self, :attribute_name => :default_value, :to => key_type.to_s.to_sym).cast!
```

Comment:
Won't it then fail with a TypeError, as per my other comment?  (Also my comments about symbol handling generally.)